test/application_system_test_case.rb

```diff
@@ -1,38 +1,42 @@
 require 'test_helper'
+require 'selenium/webdriver'
 
-DIMENSIONS = [1800, 4000].freeze
+Capybara.register_driver(:headless_chrome) do |app|
+  capabilities = Selenium::WebDriver::Remote::Capabilities.chrome(
+    chromeOptions: { args: %w[headless disable-gpu] }
+  )
 
-Capybara.register_driver :poltergeist_debug do |app|
-  Capybara::Poltergeist::Driver.new(
+  Capybara::Selenium::Driver.new(
     app,
-    phantomjs: Phantomjs.path,
-    inspector: true,
-    screen_size: DIMENSIONS,
-    window_size: DIMENSIONS
+    browser: :chrome,
+    desired_capabilities: capabilities
   )
 end
 
-Capybara.register_driver :poltergeist do |app|
-  Capybara::Poltergeist::Driver.new(
+Capybara.register_driver(:local_headless_chrome) do |app|
+  capabilities = Selenium::WebDriver::Remote::Capabilities.chrome(
+    chromeOptions: { args: %w[headless] }
+  )
+
+  Capybara::Selenium::Driver.new(
     app,
-    phantomjs: Phantomjs.path,
-    screen_size: DIMENSIONS,
-    window_size: DIMENSIONS
+    browser: :chrome,
+    desired_capabilities: capabilities
   )
 end
 
-def driven_by_default(driver = :poltergeist, using: nil)
-  driven_by driver, using: using, screen_size: DIMENSIONS
+Capybara.register_driver(:visible_chrome) do |app|
+  Capybara::Selenium::Driver.new(app, browser: :chrome)
 end
 
 class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
   case ENV['driver']
-  when 'chrome'
-    driven_by_default :selenium, using: :chrome
-  when 'poltergeist_debug'
-    driven_by_default :poltergeist_debug
+  when 'visible'
+    driven_by :visible_chrome
+  when 'faster' # maybe, don't know if gpu helps at all
+    driven_by :local_headless_chrome
```

Comment:
GPU definitely shouldn't hurt, does it cause problems on Travis? If not I would just keep it enabled and combine `headless_chrome` / `local_headless_chrome`

Reply:
but what will travis do, with gpu support? I don't know. This one I've taken over from examples as well, without actually knowing if it's correct.
I basicly put this there to try out if there is any difference. But if travis won't complain about gpu_support, I don't see a reason to have it disabled at all. So yea, would maybe make sense.